pages/fragments/header.fragment.ts

```diff
@@ -0,0 +1,52 @@
+export enum LanguageCode {
+  DE = 'de',
+  EN = 'en',
+  FR = 'fr',
+  ES = 'es',
+  NL = 'nl',
+  TR = 'tr'
+}
+
+import { Locator, Page } from "@playwright/test";
+
+export class HeaderFragment{
+    readonly page: Page;
+    readonly accountName: Locator;
+    readonly logoLink: Locator;
+    readonly navigationHome: Locator;
+    readonly navigationCategories: Locator;
+    readonly navigationContact: Locator;
+    readonly languageSelect: Locator;
+    readonly myAccount: Locator;
+    readonly myFavorites: Locator;
+    readonly myProfile: Locator;
+    readonly myInvoices: Locator;
+    readonly myMessages: Locator;
+    readonly signOut: Locator;
+
+
+    constructor(page: Page){
+        this.page = page;
+        this.accountName = page.locator('[data-test="nav-menu"]');
```

Comment:
Consider adding testIdAttribute to your config, that way, you will not have to use locator() with CSS selector, but - getByTestId method instead - https://playwright.dev/docs/api/class-testoptions#test-options-test-id-attribute

```suggestion
        this.accountName = page.getByTestId('nav-menu');
```

And let's go through all similar locators in you page objects and change them accordingly